python/nano/tutorial/notebook/training/pytorch/accelerate_pytorch_training_multi_instance.ipynb

```diff
@@ -0,0 +1,377 @@
+    "\n",
+    "    for epoch in range(num_epochs):\n",
+    "\n",
+    "        model.train()\n",
+    "        train_loss, num = 0, 0\n",
+    "        with tqdm(train_loader, unit=\"batch\") as tepoch:\n",
+    "            for data, target in tepoch:\n",
+    "                tepoch.set_description(f\"Epoch {epoch}\")\n",
+    "                optimizer.zero_grad()\n",
+    "                output = model(data)\n",
+    "                loss = loss_func(output, target)\n",
+    "                loss.backward()\n",
+    "                optimizer.step()\n",
+    "                loss_value = loss.sum()\n",
+    "                train_loss += loss_value\n",
+    "                num += 1\n",
+    "                tepoch.set_postfix(loss=loss_value)\n",
+    "            print(f'Train Epoch: {epoch}, avg_loss: {train_loss / num}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "model = MyPytorchModule()\n",
+    "optimizer = torch.optim.SGD(model.parameters(), lr=0.01, momentum=0.9, weight_decay=5e-4)\n",
+    "loss_func = torch.nn.CrossEntropyLoss()\n",
+    "train_loader = create_train_dataloader()\n",
+    "\n",
+    "training_loop(model, optimizer, train_loader, num_epochs=5, loss_func=loss_func)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; _A runnable example including this_ `training_loop` _can be seen from_ [here](https://github.com/intel-analytics/BigDL/tree/main/python/nano/tutorial/notebook/training/pytorch/accelerate_pytorch_training_multi_instance.ipynb)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> 📝 **Note**\n",
+    ">\n",
+    "> Three more parameters will be helpful if you want to customize your multi-instance training.\n",
+    "> \n",
+    "> Nano supports multiple distributed backends, i.e., ‘subprocess’ (default), ‘spawn’ (__not supported when using__ `@nano` __decorator, please subclass__ `TorchNano` __to adopt ‘spawn’__) and ‘ray’, you can switch according to your preferences by setting `distributed_backend`.\n",
+    "> \n",
+    "> By setting `num_processes`, CPU cores will be automatically and evenly distributed among specific number of processes, to avoid conflicts and maximize training throughput. If you would like to specify the CPU cores used by each process, You could set `cpu_for_each_process` to a list of length `num_processes`, in which each item is a list of CPU indices.\n",
+    "> \n",
+    "> When using data-parallel training, the batch size is equivalent to becoming `num_processes` times larger, so the learning rate warm-up strategy that gradually increase the learning rate to `num_processes` times is needed. Nano enables this practice by default through `auto_lr=True`."
+   ]
```

Comment:
Maybe this part can be adjusted a little bit (feel free to edit it if you find somewhere not reasonable):

---

```markdown
By setting `num_processes`, CPU cores will be automatically and evenly distributed among specific number of processes, to avoid conflicts and maximize training throughput. If you would like to specify the CPU cores used by each process, You could set `cpu_for_each_process` to a list of length `num_processes`, in which each item is a list of CPU indices.

Currently, `‘subprocess’` (default), and `‘ray’` are supported as `distributed_backend` for `@nano` decorator. (`'spawn'` is not supported by `@nano`)

Also note that, when using data-parallel training, the batch size is equivalent to becoming `num_processes` times larger. The learning rate warm-up strategy that gradually increases the learning rate to `num_processes` times is a compensate to achieve the same effect as single instance training. Nano enables this strategy by default through `auto_lr=True`.
```
---

As I think that the above contents are detailed enough. Maybe there is no need to link to the API doc here.